### PR TITLE
feat: Add link to pending transaction alert

### LINKED ISSUE #28308: Missing link on pending transaction alert
### What is this about?

`how to cancel or speed up a transaction` should be a hyperlink to: https://support.metamask.io/transactions-and-gas/transactions/how-to-speed-up-or-cancel-a-pending-transaction/

Screenshot 2024-11-05 at 20 26 08

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## **Description**

This PR substitutes a purely text based alert for pending transactions with one that includes a hyperlink to the docs.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28721?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/28308

## **Manual testing steps**

1. Go to this page...
2.
3.

### **After**

Screenshot 2024-12-02 at 11 13 58

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by SayaGT** [comment]
Hey Pedro let's not have a hover state for this. Let's use **primary/ defualt** for the line below to indicate it's clickable. Otherwise it looks good to me. Thanks!

> Learn how to cancel or speed up a transaction.

File: ui/components/component-library/banner-base/banner-base.types.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export interface BannerBaseStyleUtilityProps extends StyleUtilityProps {
   /**
    * The description is the content area below BannerBase title
    */
-  description?: string;
+  description?: string | React.ReactNode;
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-base/banner-base.types.ts:34]: The description is wrapped by `p` tag what are we passing here that requires a node? The banners should also accept a `children` prop which might be more suited for a custom node.

**Review by georgewrmarshall — COMMENTED** [review]
Hey @pedronfigueiredo, one question—could we also check the HTML that is rendered? I want to ensure we aren't rendering `<p>` within `<p>` and that the HTML is semantically correct. Also, could you fill out the PR description so I can better understand what this PR entails? 🙏

File: ui/pages/confirmations/hooks/alerts/transactions/PendingTransactionAlertMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      {t('pendingTransactionAlertMessage', [
+        <Text color={TextColor.primaryDefault}>
+          <a
+            href={ZENDESK_URLS.SPEEDUP_CANCEL}
+            key="link"
+            target="_blank"
+            rel="noreferrer noopener"
+          >
+            {t('pendingTransactionAlertMessageHyperlink')}
+          </a>
```
Comment (georgewrmarshall) [inline ui/pages/confirmations/hooks/alerts/transactions/PendingTransactionAlertMessage.tsx:28]: non-blocking: If you wanted to use a design system component here for consistency you could use 
```import { ButtonLink } from '../../../../../components/component-library';```
and
```suggestion
        <ButtonLink
          href={ZENDESK_URLS.SPEEDUP_CANCEL}
          key="link"
          target="_blank"
          rel="noreferrer noopener"
          >
            {t('pendingTransactionAlertMessageHyperlink')}
        </ButtonLink>
```

**Review by georgewrmarshall — DISMISSED** [review]
LGTM

**Review by OGPoyraz — DISMISSED** [review]

**Review by vinistevam — DISMISSED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**